packages/onboarding/src/step-container-v2/README.md

```diff
@@ -0,0 +1,186 @@
+
+The new version focuses on modularity and composition instead of overrides through props and CSS. We've observed that the previous version accepted a variety of props that modified how a component looks. Most of the times these properties had to play nicely with each other, which increased complexity.
+
+This version, however, provides several "slots" that allow the developer to declare _which_ parts of the step will be rendered, instead of imperatively specifying _what_ to render.
+
+Here's an example:
+
+```jsx
+import { Step } from '@automattic/onboarding';
+
+const nextButton = <Step.NextButton onClick={ navigation.submit } />
+
+<Step.StepContainerV2
+	topBar={ <Step.TopBar /> }
+	heading={ <Step.Heading text="Heading" /> }
+	stickyBottomBar={ <Step.StickyBottomBar rightButton={ nextButton } /> }
+>
+	<p>Here comes the content of the step.</p>
+	{ nextButton }
+</Step.StepContainerV2>
+```
+
+In the example above, the top bar won't have a skip nor a back button. There's a `stickyBottomBar` that will render the Next button on mobile. Although it's possible, this particular step won't render a footer.
+
+It reduces the complexity of layouting by delegating the rendering responsibility to the components within the slots. A good example of what we mean here is the `headerImageUrl` prop of `StepContainer`, which should exist if `isHorizontalLayout` is `true`, but there is no explanation of that in the code, and it's not obvious to the developer reading code introduced by others.
+
+Read the [how to extend it](#how-to-extend-it) section to learn how to reproduce this behavior in `StepContainerV2`.
+
+## How to use it
+
+**You shouldn't use `StepContainerV2` directly. Instead, use the wireframes that are exported from this package.**
+
+Aside from the stories (run `yarn storybook`), you can follow the examples from `~/client/landing/stepper/declarative-flow/internals/steps-repository`.
+
+Please do NOT override the `Step.*` components with CSS as this creates inconsistencies between steps and becomes a maintenance nightmare. Ideally, you should not need to do this as the steps are designed to be composed, and the wireframes are approved by the designers.
+
+#### What are wireframes?
+
+Wireframes are layout dispositions that were approved by the designers. They are exported from this package and can be used in the steps.
+
+## How to extend it
+
+- Try to keep the components as generic as possible, and without adding overrides to components coming from `@wordpress/components` unless explicitly approved by the designers.
+- If only one step uses something, it probably means it shouldn't belong to `StepContainerV2`. Try using some of the slots, or add a new one.
```

Comment:
If the step has a unique UI element that does not exist in other steps, it probably does not belong in `StepContainerV2`. Try using some of the slots, or add a new one.